Tests/OptimizelyTests-APIs/OptimizelyClientTests_OptimizelyJSON.swift

```diff
@@ -0,0 +1,510 @@
+    func testToMap() {
+        var err: Error?
+        var optimizelyJSONToMap: [String:Any]!
+        do {
+            try optimizelyJSONToMap = self.optimizelyJSON.toMap()
+        } catch {
+            err = error
+        }
+        XCTAssertNil(err)
+        XCTAssertTrue(NSDictionary(dictionary: optimizelyJSONToMap).isEqual(to: self.data))
+    }
+    
+    func testToString() {
+        var err: Error?
+        var optimizelyJSONToString: String!
+        do {
+            try optimizelyJSONToString = self.optimizelyJSON.toString()
+        } catch {
+            err = error
+        }
+        XCTAssertNil(err)
+        XCTAssertEqual(optimizelyJSONToString, self.payload)
+    }
+    
+    func testGetValueForInvalidJSONKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4.", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.valueForKeyNotFound("").reason)
+    }
+    
+    func testGetValueForMissingJSONKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "some_key", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.valueForKeyNotFound("some_key").reason)
+    }
+    
+    func testGetValueForInvalidJSONMultipleKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field3.some_key", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.valueForKeyNotFound("some_key").reason)
+    }
+    
+    func testGetValueForValidJSONKeyAndEmptySchema() {
+        var err: Error?
+        var schema = EmptySchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertNil(err)
+    }
+    
+    func testGetValueForValidJSONKeyAndNonDecodableEmptySchema() {
+        var err: Error?
+        var schema = NonDecodableSchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.failedToAssignValueToSchema.reason)
+    }
+    
+    func testGetValueForValidJSONKeyAndIncorrectDecodableStructSchema() {
+        var err: Error?
+        var schema = IncorrectSchema()
+        do {
+            try self.optimizelyJSON.getValue(jsonPath: "field4", schema: &schema)
+        } catch {
+            err = error
+        }
+        XCTAssertEqual(err?.localizedDescription, OptimizelyError.failedToAssignValueToSchema.reason)
```

Comment:
I don't think it should fail. if the user provides an empty or incorrect schema , it will not be parsed correctly but I don't think it should fail.

Reply:
It fails since `IncorrectSchema` has a mandatory property defined as `var incorrectField: Int = 0`, On the other hand it will pass if it was defined as optional  `var incorrectField: Int?`.

Reply:
We have defined `map` as `[String: Any]` which makes sure that values are non-nil hence i couldn't write test for `field6`, should we make them nullable? is there a case where a json string will have a `nil` value?

Reply:
Adding a test for `field4..some_field` 👍